tests/PasswordTest.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace atk4\data\tests;
+
+use atk4\data\Model;
+use atk4\data\Persistence_Array;
+
+class PasswordTest extends TestCase
+{
+    public function testPasswordField()
+    {
+        $m = new Model(); //$db, 'job');
+
+        $m->addField('p', ['Password']);
+
+        $m['p'] = 'mypass';
+
+        // when setting password, you cannot retrieve it back
+        $this->assertEquals('mypass', $m['p']);
+
+        // password changed, so it's dirty.
+        $this->assertEquals(true, $m->isDirty('p'));
+
+        $this->assertEquals(false, $m->compare('p', 'badpass'));
+        $this->assertEquals(true, $m->compare('p', 'mypass'));
+    }
+
+    public function testPasswordPersistence1()
+    {
+        $a = [];
+        $p = new Persistence_Array($a);
+        $m = new Model($p);
+
+        $m->addField('p', ['Password']);
+
+        $m['p'] = 'mypass';
+
+        $this->assertEquals('mypass', $m['p']);
```

Comment:
why this don't fail ?

Reply:
same as above - until you save password it will work like this.